python/pyspark/sql/session.py

```diff
@@ -1630,6 +1630,13 @@ def sql(
         -------
         :class:`DataFrame`
 
+        Notes
+        -----
+        In Classic Spark, a referenced temporary view is resolved immediately, while in Spark
+        Connect it is lazy evaluated.
+        So in Spark Connect if a view is dropped, modified or replaced after `spark.sql`, the
+        execution may fail or generate different results.
```

Comment:
Out of cusiority, in which cases the execution may fail?

Reply:
drop the view, for example
```
df = ...
df.createTempView("some_view")
df2 = spark.sql("SELECT * FROM some_view")
spark.catalog.dropTempView("some_view") <- drop the view
df2.show() <- should fail in Spark Connect
```